controller/aks-cluster-config-handler.go

```diff
@@ -0,0 +1,891 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	"reflect"
+	"strings"
+	"time"
+
+	containersv209 "github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2020-09-01/containerservice"
+	containersv211 "github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2020-11-01/containerservice"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-10-01/resources"
```

Comment:
`resorces` SDK is needed for resource groups
`containerservices` SDK is for creating AKS cluster 
we need two different versions of `containerservices` because they offer different features, which is weird but `.../2020-09-01/containerservice` contains ListOperators function and `2020-11-01/containerservice` has NetworkProfile definition

Reply:
This is curiosity point to follow-up on @thedadams point: these all share the same Go module version, I assume, but have different paths within that module?

Reply:
Azure does interesting things with their SDK to make it so you can update it to use new features while maintaining compatibility with code that uses old features. I believe the ideal for a new project would be to not use multiple versions as it gives you the opportunity to leverage just the newest or a newer package. Maybe it is not organized the way I remember but we shouldn't be doing anything a legacy way if there is a way to achieve the same thing with a newer package we are importing.

Reply:
It is the way how Azure API is developed, which reminds me OpenStack Heat API versioning scheme.